Classes/Service/XsdGenerator.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace SMS\FluidComponents\Service;
+
+use Sitegeist\FluidStyleguide\Domain\Model\Component;
+use Sitegeist\FluidStyleguide\Domain\Model\ComponentName;
+use SMS\FluidComponents\Fluid\ViewHelper\ComponentRenderer;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3Fluid\Fluid\Core\ViewHelper\ArgumentDefinition;
+
+class XsdGenerator
+{
+    /**
+     * @var object|\SMS\FluidComponents\Utility\ComponentLoader
+     */
+    private $componentLoader;
+
+    public function __construct()
+    {
+        $this->componentLoader = GeneralUtility::makeInstance(\SMS\FluidComponents\Utility\ComponentLoader::class);
```

Comment:
Können wir hier Dependency Injection verwenden, z. B. im Constructor?

Reply:
Klar, habe ich umgebaut. Aber im command dann hart verdrahtet, denn bei TYPO3 commands lässt sich keine Constructor DI verwenden, da die CommandRegistry den Constructor mit exakt 1 Parameter aufruft:  $this->commands[$commandName] = GeneralUtility::makeInstance($commandConfig['class'], $commandName);